### DISCUSSION TITLE
Questions about the nomination and use of data sources

### DISCUSSION THREAD

**Discussion by xdxTao** [osm-search/Nominatim#3434] [Q&A]
Hello, I have built a service on my own server, and the imported data comes from https://download.geofabrik.de/asia/china.html

After testing, I found that the data is incomplete. The query results of the service I built are different from those from the Nominatim API. The results of Nominatim are more accurate, so I want to know where Nominatim's data source is downloaded from. If possible, I hope to tell me the download address of China's data directly, thank you very much

**Comment by mtmail** [comment]
Both use the same OpenStreetMap data. Can you give examples where your Nominatim installation returns different results than nominatim.openstreetmap.org?

**Reply by xdxTao** [reply]
This is the request and result of the self-built service

`{
    "type": "FeatureCollection",
    "geocoding": {
        "version": "0.1.0",
        "attribution": "Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
        "licence": "ODbL",
        "query": "拉萨市"
    },
    "features": [
        {
            "type": "Feature",
            "properties": {
                "geocoding": {
                    "place_id": 126888,
                    "osm_type": "relation",
                    "osm_id": 2745046,
                    "osm_key": "boundary",
                    "osm_value": "administrative",
                    "type": "city",
                    "label": "拉萨市, 850000, 中国",
                    "name": "拉萨市",
                    "country": "中国",
                    "postcode": "850000",
                    "admin": {
                        "level5": "拉萨市"
                    }
                }
            },
            "geometry": {
                "type": "Point",
                "coordinates": [
                    91.1173015,
                    29.6542054
                ]
            }
        }
    ]
}`

This is the request and result of Nominatim

`{
    "type": "FeatureCollection",
    "geocoding": {
        "version": "0.1.0",
        "attribution": "Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright",
        "licence": "ODbL",
        "query": "拉萨市"
    },
    "features": [
        {
            "type": "Feature",
            "properties": {
                "geocoding": {
                    "place_id": 220815117,
                    "osm_type": "relation",
                    "osm_id": 2745046,
                    "osm_key": "boundary",
                    "osm_value": "administrative",
                    "type": "city",
                    "label": "拉萨市, 西藏自治区, 850000, 中国",
                    "name": "拉萨市",
                    "postcode": "850000",
                    "state": "西藏自治区",
                    "country": "中国",
                    "country_code": "cn",
                    "admin": {
                        "level5": "拉萨市",
                        "level4": "西藏自治区"
                    }
                }
            },
            "geometry": {
                "type": "Point",
                "coordinates": [
                    91.1173015,
                    29.6542054
                ]
            }
        }
    ]
}`


The self-built service is missing a level 4 layer. All the queries under this are indeed missing this level. I use "osm_id": 2745046 to query my self-built database and there is no data, so I suspect that the data downloaded from https://download.geofabrik.de/asia/china.html is incomplete.

**Reply by xdxTao** [reply]
Sorry, forgot to post the request link above

http://10.0.1.18:8090/search/search?q=拉萨市&format=geocodejson&addressdetails=1&accept-language=zh
https://nominatim.openstreetmap.org/search?q=拉萨市&format=geocodejson&addressdetails=1&accept-language=zh


Looking forward to your reply

**Reply by mtmail** [reply]
Indeed the state 'Tibet' is incomplete in the China file on download.geofabrik.de. It's not related to Nominatim software.

In OpenStreetMap data schema a relation (here https://www.openstreetmap.org/relation/153292) is made up of multiple ways and each way is made up of many nodes. https://wiki.openstreetmap.org/wiki/Elements

The China file you downloaded contains the relation, but it's missing 17 ways. So Nominatim didn't import the relation.

```bash
wget https://download.geofabrik.de/asia/china-latest.osm.pbf

# https://osmcode.org/osmium-tool/
# https://www.openstreetmap.org/relation/913012
osmium getid --add-referenced --output jiangsu.osm china-latest.osm.pbf r913012
# https://www.openstreetmap.org/relation/153292
osmium getid --add-referenced --output tibet.osm china-latest.osm.pbf r153292

osmium check-refs --check-relations jiangsu.osm
There are 248152 nodes, 9297 ways, and 3089 relations in this file.
Nodes     in ways      missing: 0
Nodes     in relations missing: 0
Ways      in relations missing: 0
Relations in relations missing: 0

osmium check-refs --check-relations tibet.osm
There are 98570 nodes, 2463 ways, and 778 relations in this file.
Nodes     in ways      missing: 0
Nodes     in relations missing: 0
Ways      in relations missing: 17 <===
Relations in relations missing: 0
```

You can contact Geofabrik asking to improve the extraction process. Basically once per night they take the global OpenStreetMap data (called the planet https://planet.openstreetmap.org/) and cut out countries into individual files. Sometimes that's not precise enough or borders changed. It's possible the borders changed only a couple of metres.

You could download all of Asia and extract the needed regions yourself.

You can manually add the relation with all ways to your Nominatim installation :
```
cd $nominatim-project-directory
nominatim add-data --relation 153292
nominatim index
```

If that didn't work try to reindex everything in that state.
```
nominatim refresh --data-area r153292
nominatim index
```
